docs/README.md

```diff
@@ -31,15 +31,41 @@ whichever version of Spark you currently have checked out of revision control.
 The Spark documentation build uses a number of tools to build HTML docs and API docs in Scala, Java,
 Python, R and SQL.
 
-You need to have [Ruby](https://www.ruby-lang.org/en/documentation/installation/) and
-[Python](https://docs.python.org/2/using/unix.html#getting-and-installing-the-latest-version-of-python)
-installed. Also install the following libraries:
+You need to have Ruby 2 and Python 3 installed. A handy way to install and manage various versions of Ruby and Python is with [`rbenv`] and [`pyenv`].
+
+[`rbenv`]: https://github.com/rbenv/rbenv
+[`pyenv`]: https://github.com/pyenv/pyenv
+
+On macOS you can install them with Homebrew:
+
+```sh
+brew install rbenv pyenv
+```
+
+To activate them, you'll need to run these commands or add them to the end of your `.bash_profile`:
+
+```sh
+eval "$(rbenv init -)"
+eval "$(pyenv init -)"
```

Comment:
One alternative to running these `init` commands is to prefix the below commands so that they do the right thing without depending on any environment variables being set by `init`.

So, for example, `pip install` would become `pyenv exec pip install` and `jekyll` would become `rbenv exec jekyll`.